cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/cache/DynamicDatasetCacheTest.java

```diff
@@ -140,9 +154,10 @@ protected void testDatasetCache(@Nullable Map<String, TestDataset> datasetMap)
     // verify that static datasets are in the tx-awares after start
     TransactionContext txContext = cache.newTransactionContext();
     txAwares = getTxAwares();
-    Assert.assertEquals(2, txAwares.size());
+    Assert.assertEquals(3, txAwares.size());
     Assert.assertSame(a, txAwares.get(0));
-    Assert.assertSame(b, txAwares.get(1));
+    Assert.assertSame(a_n2, txAwares.get(1));
+    Assert.assertSame(b, txAwares.get(2));
```

Comment:
I look into the code. The txAwares is implemented by a `TreeSet` which ensures the values are ordered by key. This is why it works. 
I agree it is brittle as the implementation might change. I now use `List.Contains` to do the check.